src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java

```diff
@@ -451,6 +507,22 @@ private Set<PaymentInstructionInformation3> constructPaymentInstructionInformati
         return paymentInstructionInformationSet;
     }
 
+    /*
+     * CU Customization: Added helper method for determining expected Immediate Payment flag based on payment urgency.
+     */
+    private boolean getExpectedProcessImmediateFlag(
+            final ExtractTypeContext extractTypeContext
+    ) {
+        if (extractTypeContext.isLimitedToPaymentsWithUrgency(PaymentUrgency.REGULAR)) {
+            return false;
+        } else if (extractTypeContext.isLimitedToPaymentsWithUrgency(PaymentUrgency.IMMEDIATE)) {
+            return true;
+        } else {
+            throw new IllegalArgumentException("Context of type " + extractTypeContext.getClass().getName()
```

Comment:
Would it be better to default to false rather than error?

Reply:
The enclosing `getExpectedProcessImmediateFlag()` method and the context's new `isLimitedToPaymentsWithUrgency()` method are only meant to be invoked when processing Checks. If an explicit Payment Urgency setting cannot be discerned in this situation, I would prefer to throw an exception here, since such a situation suggests a misconfiguration or other coding error.

If needed, I could rename the `getExpectedProcessImmediateFlag()` method, to make it clearer that it's only intended for Check processing situations.

Reply:
I'm renaming the get-expected-flag method to make its intent clearer. If you still think the method should no longer have the exception-throwing part, please let me know.

Reply:
renaming sounds good, thanks